### ISSUE TITLE
Getting the parse error into a variable

### ISSUE THREAD

**Issue by manisshaa** [cucumber/cucumber-js#1404]
If the feature file is not in proper gherkin format, Cucumber shows a parse error on console and aborts the tests. Following is the code that does that (this is a snippet from cucumber > lib > cli > run.js

function exitWithError(error) {
  console.error(_verror.default.fullStack(error)); // eslint-disable-line no-console
  process.exit(1);
}

I want to capture this parse error into a variable or a file. As a work around, I have added a line to function exitWithError to write the parse error to a file for me to later read the error for reporting purposes. But this solution requires my own version of cucumber package and then use the same. But the problem is that my changes will be overwritten once cucumber package is updated. 

function exitWithError(error) {
  fs.writeFileSync("./Parse_Error", error);
  console.error(_verror.default.fullStack(error)); // eslint-disable-line no-console
  process.exit(1);
}

Please suggest.

Note: Please know that I need to handle this scenario as we don't manually create cucumber feature files but import them from Jira cucumber tests. These tests are created by stakeholders who may make some minor mistakes in these cucumber tests which may lead to the parse error. I need to capture the parse error (in a file or variable), so that I can report it back to the jira ticket as a comment or something

**Comment by davidjgoss** [comment]
The messages protocol [includes a `ParseError` message type](https://github.com/cucumber/cucumber/pull/969), so if we start emitting these before we exit then we should be able to pick up from message formatter output. I'll look at doing this change before 7.0.0

**Comment by davidjgoss** [comment]
This change will be available in the forthcoming 8.0.0 release.

The simplest way to capture the parse errors will be to direct the [message formatter](https://github.com/cucumber/cucumber-js/blob/main/docs/formatters.md#message) output to a file, like:

```shell
npx cucumber-js --format message:cucumber-messages.ndjson
```

You'll then be able to see `parseError` messages in that file.

For more programmatic control, you could [implement a custom formatter](https://github.com/cucumber/cucumber-js/blob/main/docs/custom_formatters.md) which listens to the emitted messages and then acts on the `parseError` ones, something like:


```js
import {Formatter} from '@cucumber/cucumber`

class MyFormatter extends Formatter {
  constructor(options) {
    super(options)
    options.eventBroadcaster.on('envelope', (envelope) => {
      if (envelope.parseError) {
        // do something with parse error
      }
    }
  }
}
```
